go/vt/vtgate/planbuilder/route_planning.go

```diff
@@ -708,44 +803,53 @@ func createRoutePlan(table *abstract.QueryTable, solves semantics.TableSet, rese
 	default:
 		plan.routeOpCode = engine.SelectScatter
 	}
-	err = plan.addPredicate(table.Predicates...)
+	err = plan.addPredicate(ctx, table.Predicates...)
 	if err != nil {
 		return nil, err
 	}
 
 	return plan, nil
 }
 
-func findColumnVindex(a *routeTree, exp sqlparser.Expr, sem *semantics.SemTable) vindexes.SingleColumn {
-	left, isCol := exp.(*sqlparser.ColName)
+func findColumnVindex(ctx optimizeContext, a *routeTree, exp sqlparser.Expr) vindexes.SingleColumn {
+	_, isCol := exp.(*sqlparser.ColName)
 	if !isCol {
 		return nil
 	}
-	leftDep := sem.BaseTableDependencies(left)
 
 	var singCol vindexes.SingleColumn
 
-	_ = visitRelations(a.tables, func(rel relation) (bool, error) {
-		rb, isRoute := rel.(*routeTable)
-		if !isRoute {
-			return true, nil
-		}
-		if leftDep.IsSolvedBy(rb.qtable.TableID) {
-			for _, vindex := range rb.vtable.ColumnVindexes {
-				sC, isSingle := vindex.Vindex.(vindexes.SingleColumn)
-				if isSingle && vindex.Columns[0].Equal(left.Name) {
-					singCol = sC
-					return false, io.EOF
+	for _, expr := range ctx.semTable.GetExprAndEqualities(exp) {
```

Comment:
needs comments that explain wth is going on

Reply:
fixed in 710f421513